src/utils/application.data.ts

```diff
@@ -2,34 +2,64 @@ import { Session } from '@companieshouse/node-session-handler';
 import { Request } from "express";
 
 import { createAndLogErrorRequest } from './logger';
-import { ID } from '../model/data.types.model';
+import { ID, OverseasEntityKey, Transactionkey } from '../model/data.types.model';
+
 import {
   ApplicationData,
   APPLICATION_DATA_KEY,
   ApplicationDataType,
   ApplicationDataArrayType
 } from "../model";
+
 import { BeneficialOwnerGov, BeneficialOwnerGovKey } from '../model/beneficial.owner.gov.model';
 import { BeneficialOwnerIndividual, BeneficialOwnerIndividualKey } from '../model/beneficial.owner.individual.model';
 import { BeneficialOwnerOtherKey } from '../model/beneficial.owner.other.model';
 import { ManagingOfficerCorporate, ManagingOfficerCorporateKey } from '../model/managing.officer.corporate.model';
 import { ManagingOfficerIndividual, ManagingOfficerKey } from '../model/managing.officer.model';
-import { PARAM_BENEFICIAL_OWNER_GOV, PARAM_BENEFICIAL_OWNER_INDIVIDUAL, PARAM_BENEFICIAL_OWNER_OTHER, PARAM_MANAGING_OFFICER_CORPORATE, PARAM_MANAGING_OFFICER_INDIVIDUAL } from '../config';
+
+import {
+  FEATURE_FLAG_ENABLE_REDIS_REMOVAL,
+  PARAM_BENEFICIAL_OWNER_GOV,
+  PARAM_BENEFICIAL_OWNER_INDIVIDUAL,
+  PARAM_BENEFICIAL_OWNER_OTHER,
+  PARAM_MANAGING_OFFICER_CORPORATE,
+  PARAM_MANAGING_OFFICER_INDIVIDUAL,
+  ROUTE_PARAM_TRANSACTION_ID,
+  ROUTE_PARAM_SUBMISSION_ID
+} from '../config';
+
 import { BeneficialOwnerCorporate } from '@companieshouse/api-sdk-node/dist/services/overseas-entities';
 import { Remove } from 'model/remove.type.model';
+import { isActiveFeature } from "./feature.flag";
+import { getOverseasEntity } from "../service/overseas.entities.service";
+
+export const getApplicationData = async (session: Session | undefined, req?: Request): Promise<ApplicationData> => {
```

Comment:
I assume at some point this function will no longer take a `session` as parameter? Retrieval will always be based on a transaction and submission id?

Reply:
Absolutely. When we have an entirely REDIS-free journey, this method will need not have this parameter.